### PR TITLE
Update UpgradeIntegrationTest to use n-3 to current Marathon versions

### PR DESCRIPTION
**Pull request by timcharper**
- Instead of hard-coding so many lines to a specific version of Marathon, we change the test to talk about n-3 to n-1 versions of Marathon
- Test suite fails if n-1 minor version is current minor version - 1, helping us to remember to keep this test updated in the future

JIRA Issues: MARATHON-8703

### CODE REVIEW

File: tests/integration/src/test/scala/mesosphere/marathon/integration/UpgradeIntegrationTest.scala [hunk 1]
```diff
@@ -32,72 +31,57 @@ class UpgradeIntegrationTest extends AkkaIntegrationTest with MesosClusterTest w
 
   val zkURLBase = s"zk://${zkServer.connectUri}/marathon-$suiteName"
 
-  val marathon149Artifact = MarathonArtifact("1.4.9", "marathon-1.4.9.tgz")
-  val marathon1515Artifact = MarathonArtifact("1.5.15", "marathon-1.5.15.tgz")
-  val marathon16549Artifact = MarathonArtifact("1.6.549", "marathon-1.6.549-aabf74302.tgz")
+  val marathonMinus3Artifact = MarathonArtifact(SemVer(1, 6, 567, Some("2d8b3e438")))
```
Comment (jeschkies) [inline tests/integration/src/test/scala/mesosphere/marathon/integration/UpgradeIntegrationTest.scala]: This one is tricky. We used 1.4.9 and 1.5.15 to guarantee that we can still consume the task ids created by these versions. That was the original idea of the test. Can we produce Mesos tasks started by Marathon 1.4.9 and 1.5.15 without using these?

Comment (timcharper) [reply tests/integration/src/test/scala/mesosphere/marathon/integration/UpgradeIntegrationTest.scala]: I think I'll feel okay with unit tests to assert this going forward

Comment (jeschkies) [reply tests/integration/src/test/scala/mesosphere/marathon/integration/UpgradeIntegrationTest.scala]: I'm not aware of any unit test that verifies task id parsing for all possible ids and labels.

File: src/test/scala/mesosphere/marathon/core/task/TaskTest.scala [hunk 2]
```diff
@@ -247,4 +246,68 @@ class TaskTest extends UnitTest with Inside {
     }
   }
 
+  val marathon_1_4_9_ephemeral_app_id = "sleeper.7dcb9c7f-e01d-11e9-80dd-2ae4d7ed9076"
+  val marathon_1_4_9_ephemeral_pod_id = "dev_sleeper.instance-71471cee-e01d-11e9-80dd-2ae4d7ed9076.sleep1"
+  val marathon_1_4_9_resident_app_id = "dev_resident-sleep.d15865ae-e01f-11e9-8fd4-2ae4d7ed9076.2"
+  "task ID parsing" should {
```
Comment (jeschkies) [inline src/test/scala/mesosphere/marathon/core/task/TaskTest.scala]: Awesome. I just wanted to point you to `Task.Id.parse`. Thanks for going the extra mile!

**Pull request merged**